src/components/Counter/Counter.js

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { getParentForm } from '../../util';
+
+class Counter extends React.Component {
+  constructor() {
+    super();
+
+    this.handleIncrement = this.handleIncrement.bind(this);
+    this.handleAddItem = this.handleAddItem.bind(this);
+    this.handleToggleAddForm = this.handleToggleAddForm.bind(this);
+
+    this.state = {
+      addFormShown: false,
+    };
+  }
+
+  handleIncrement(itemId) {
+    return () => {
```

Comment:
Handlers shouldn't return functions, they should be able to be called directly. The accepted pattern is:

```
<button onClick={() => this.handleRemoveItem(item.id)}>✖</button>
```